packages/wrangler/src/dev/use-esbuild.ts

```diff
@@ -199,11 +199,23 @@ export function useEsbuild({
 			// Capture the `stop()` method to use as the `useEffect()` destructor.
 			stopWatching = bundleResult?.stop;
 
-			// if "noBundle" is true, then we need to manually watch the entry point and
-			// trigger "builds" when it changes
+			// if "noBundle" is true, then we need to manually watch all modules and
+			// trigger "builds" when any change
 			if (noBundle) {
-				const watcher = watch(entry.file, {
+				const watching = [path.resolve(entry.moduleRoot)];
+				// Should we try and watch a Python reqyurements.txt file?
+				const pythonRequirements =
+					getBundleType(entry.format, entry.file) === "python"
+						? path.resolve(entry.directory, "requirements.txt")
+						: undefined;
+
+				if (pythonRequirements) {
+					watching.push(pythonRequirements);
+				}
+
+				const watcher = watch(watching, {
 					persistent: true,
+					ignored: [".git", "node_modules"],
```

Comment:
Should we be pulling in the `.gitignore` file here, perhaps?

I always get a bit wary when this stuff is hard-coded, as has been noted recently in Pages where it refuses to upload files within a node_modules folder.

Reply:
This was intended to mimic the behaviour in https://github.com/cloudflare/workers-sdk/blob/d67836a599d55e5f0932cc5f1cc901e640015f47/packages/wrangler/src/deployment-bundle/find-additional-modules.ts#L177-L193

Reply:
LOL which is code I wrote 😆 - OK, given that this is already the behaviour, I guess we are safe.